packages/editor/src/components/preview-dropdown/index.js

```diff
@@ -62,33 +65,88 @@ export default function PreviewDropdown( { forceIsAutosaveable, disabled } ) {
 		desktop,
 	};
 
+	const isZoomOutExperiment =
+		!! window.__experimentalEnableZoomedOutPatternsTab;
+
+	const zoomIcon = () => {
+		if ( __unstableGetEditorMode() === 'zoom-out' ) {
+			return __( '50%' );
+		}
+		return __( '100%' );
+	};
+
+	const getIcon = () => {
+		if ( isZoomOutExperiment && deviceType === 'Desktop' ) {
+			return zoomIcon;
+		}
+		return deviceIcons[ deviceType.toLowerCase() ];
+	};
+
 	return (
 		<DropdownMenu
 			className="editor-preview-dropdown"
 			popoverProps={ popoverProps }
 			toggleProps={ toggleProps }
 			menuProps={ menuProps }
-			icon={ deviceIcons[ deviceType.toLowerCase() ] }
+			icon={ getIcon() }
 			label={ __( 'View' ) }
 			disableOpenOnArrowDown={ disabled }
 		>
 			{ ( { onClose } ) => (
 				<>
 					<MenuGroup>
 						<MenuItem
-							onClick={ () => setDeviceType( 'Desktop' ) }
-							icon={ deviceType === 'Desktop' && check }
+							onClick={ () => {
+								if ( isZoomOutExperiment ) {
+									// Rather than setting to edit, we may need to set back to whatever it was previously.
+									__unstableSetEditorMode( 'edit' );
+								}
+								setDeviceType( 'Desktop' );
+							} }
+							icon={
+								deviceType === 'Desktop' &&
+								__unstableGetEditorMode() !== 'zoom-out' &&
+								check
+							}
 						>
-							{ __( 'Desktop' ) }
+							{ isZoomOutExperiment
+								? __( 'Zoom to 100%' )
+								: __( 'Desktop' ) }
 						</MenuItem>
+						{ isZoomOutExperiment && (
+							<MenuItem
+								onClick={ () => {
+									setDeviceType( 'Desktop' );
+									__unstableSetEditorMode( 'zoom-out' );
+								} }
+								icon={
+									__unstableGetEditorMode() === 'zoom-out' &&
+									check
+								}
+							>
+								{ __( 'Zoom to 50%' ) }
+							</MenuItem>
+						) }
 						<MenuItem
-							onClick={ () => setDeviceType( 'Tablet' ) }
+							onClick={ () => {
+								setDeviceType( 'Tablet' );
+								if ( isZoomOutExperiment ) {
+									// Rather than setting to edit, we may need to set back to whatever it was previously.
+									__unstableSetEditorMode( 'edit' );
+								}
```

Comment:
The non-zoom out ones are all similar: Set device type + exit zoom out. I think a named function to do this would make the code easier: `handleDevicePreviewClick( deviceType, zoomOut = false )`